pkg/quotaplugins/quota-forest/quota-manager/quota/quotamanager.go

```diff
@@ -369,33 +369,40 @@ func (m *Manager) UpdateForest(forestName string) (unallocatedConsumerIDs []stri
 // AllocateForest : allocate a consumer on a forest
 func (m *Manager) AllocateForest(forestName string, consumerID string) (response *core.AllocationResponse, err error) {
 	forestController := m.forests[forestName]
+	var awExistsInNamespace bool = false
 	if forestController == nil {
 		return nil, fmt.Errorf("invalid forest name %s", forestName)
 	}
 	consumerInfo := m.consumerInfos[consumerID]
 	if consumerInfo == nil {
 		return nil, fmt.Errorf("consumer %s does not exist, create and add first", consumerID)
 	}
-	if forestController.IsConsumerAllocated(consumerID) {
-		//allocation can pass if consumers are not released
-		//make sure that appwrapper (consumer) does not exist in target namespace
-		re := regexp.MustCompile("_")
-		split := re.Split(consumerID, -1)
-		namespace := split[1]
-		awName := split[3]
-		config, err := rest.InClusterConfig()
-		if err != nil {
-			klog.Errorf("unable to retrieve in-cluster kubeconfig %v", err)
-		}
-		arbclients := clientset.NewForConfigOrDie(config)
-		_, err = arbclients.ArbV1().AppWrappers(namespace).Get(awName, v1.GetOptions{})
-		if err != nil {
+	//allocation can pass if consumers are not released
+	//make sure that appwrapper (consumer) does not exist in target namespace
+	re := regexp.MustCompile("_")
+	split := re.Split(consumerID, -1)
```

Comment:
I realized something. This may be a problem later. If the namespace has an underscore in the name, your indexing would fail. There is a function called `ParseId` in `./pkg/quotaplugins/util/utils.go` that does this for you and returns the namespace and the AppWrapper. Let's switch to that.

Reply:
Thanks, done